src/main/java/org/gridsuite/sensitivityanalysis/server/service/SensitivityAnalysisObserver.java

```diff
@@ -0,0 +1,83 @@
+import lombok.NonNull;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+/**
+ * @author Florent MILLOT <florent.millot at rte-france.com>
+ */
+@Service
+public class SensitivityAnalysisObserver {
+
+    private static final String OBSERVATION_PREFIX = "app.computation.";
+    private static final String PROVIDER_TAG_NAME = "provider";
+    private static final String TYPE_TAG_NAME = "type";
+    private static final String STATUS_TAG_NAME = "status";
+    private static final String COMPUTATION_TYPE = "sensi";
+    private static final String COMPUTATION_COUNTER_NAME = OBSERVATION_PREFIX + "count";
+
+    private final String defaultProvider;
+    private final ObservationRegistry observationRegistry;
+    private final MeterRegistry meterRegistry;
+
+    public SensitivityAnalysisObserver(@Value("${sensitivity-analysis.default-provider}") String defaultProvider,
+                                       @NonNull ObservationRegistry observationRegistry,
+                                       @NonNull MeterRegistry meterRegistry) {
+        this.defaultProvider = defaultProvider;
+        this.observationRegistry = observationRegistry;
+        this.meterRegistry = meterRegistry;
+    }
+
+    public <E extends Throwable> void observe(String name, SensitivityAnalysisRunContext runContext, Observation.CheckedRunnable<E> runnable) throws E {
+        createObservation(name, runContext).observeChecked(runnable);
+    }
+
+    public <T, E extends Throwable> T observe(String name, SensitivityAnalysisRunContext runContext, Observation.CheckedCallable<T, E> callable) throws E {
+        return createObservation(name, runContext).observeChecked(callable);
+    }
+
+    public <T extends SensitivityAnalysisResult, E extends Throwable> T observeRun(String name, SensitivityAnalysisRunContext runContext, Observation.CheckedCallable<T, E> callable) throws E {
+        T result = createObservation(name, runContext).observeChecked(callable);
+        incrementCount(runContext, result);
+        return result;
+    }
+
+    private Observation createObservation(String name, SensitivityAnalysisRunContext runContext) {
+        String provider = runContext.getProvider() != null ? runContext.getProvider() : defaultProvider;
+        return Observation.createNotStarted(OBSERVATION_PREFIX + name, observationRegistry)
+            .lowCardinalityKeyValue(PROVIDER_TAG_NAME, provider)
+            .lowCardinalityKeyValue(TYPE_TAG_NAME, COMPUTATION_TYPE);
+    }
+
+    private void incrementCount(SensitivityAnalysisRunContext runContext, SensitivityAnalysisResult result) {
+        String provider = runContext.getProvider() != null ? runContext.getProvider() : defaultProvider;
+        Counter.builder(COMPUTATION_COUNTER_NAME)
+            .tag(PROVIDER_TAG_NAME, provider)
+            .tag(TYPE_TAG_NAME, COMPUTATION_TYPE)
+            .tag(STATUS_TAG_NAME, getStatusFromResult(result))
```

Comment:
Remove this

Reply:
You mean I should not use incrementCount as in security server ?

Reply:
As discussed, I always return OK